library/core/src/ptr/mod.rs

```diff
@@ -791,8 +800,11 @@ pub const fn from_ref<T: ?Sized>(r: &T) -> *const T {
 
 /// Convert a mutable reference to a raw pointer.
 ///
-/// This is equivalent to `r as *mut T`, but is a bit safer since it will never silently change
-/// type or mutability, in particular if the code is refactored.
+/// The caller must ensure that the pointee outlives the pointer this function returns, or else it
+/// will end up pointing to garbage.
```

Comment:
Same changes here: avoid "garbage" and also add an example of what to avoid.

Also, the structure of this comment should be made symmetric with the one for `from_ref`; i.e. move "The caller must ..." below the next paragraph.